test/built-ins/AsyncContext/context-propagation/generators-yield-inside-run.js

```diff
@@ -0,0 +1,33 @@
+// Copyright (C) 2023 Igalia, S. L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: TODO
+features: [AsyncContext]
+---*/
+
+// TODO: This test tests the behavior in
+// https://github.com/tc39/proposal-async-context/pull/61
+
+const asyncVar = new AsyncContext.Variable();
+
+function* gen() {
+  assert.sameValue(asyncVar.get(), "init");
```

Comment:
This assert is not reachable in the "init" run, because the generator is in the SUSPENDED-START state until the "first-iter" run.

We can either remove it, or add a generator.next() call to the "init" run invocation.